controllers/resources/authconfig.go

```diff
@@ -0,0 +1,172 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	_ "embed" // needed for go:embed directive
+	"strings"
+
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	authorinov1beta2 "github.com/kuadrant/authorino/api/v1beta2"
+	"github.com/opendatahub-io/odh-model-controller/controllers/utils"
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AuthType string
+
+const (
+	UserDefined AuthType = "userdefined"
+	Anonymous   AuthType = "anonymous"
+)
+
+type AuthConfigTemplateLoader interface {
+	Load(authType AuthType, key types.NamespacedName) (authorinov1beta2.AuthConfig, error)
+}
+
+type AuthTypeDetector interface {
+	Detect(ctx context.Context, isvc *kservev1beta1.InferenceService) (AuthType, error)
+}
+
+type AuthConfigStore interface {
+	Get(ctx context.Context, key types.NamespacedName) (authorinov1beta2.AuthConfig, error)
+	Remove(ctx context.Context, key types.NamespacedName) error
+	Store(ctx context.Context, authConfig authorinov1beta2.AuthConfig) error
+}
+
+//go:embed template/authconfig_anonymous.yaml
+var authConfigTemplateAnonymous []byte
+
+//go:embed template/authconfig_userdefined.yaml
+var authConfigTemplateUserDefined []byte
+
+type staticTemplateLoader struct {
+}
+
+func NewStaticTemplateLoader() AuthConfigTemplateLoader {
+	return &staticTemplateLoader{}
+}
+
+func (s *staticTemplateLoader) Load(authType AuthType, key types.NamespacedName) (authorinov1beta2.AuthConfig, error) {
+	authConfig := authorinov1beta2.AuthConfig{}
+	if authType == UserDefined {
+		err := utils.ConvertToStructuredResource(authConfigTemplateUserDefined, &authConfig)
+		if err != nil {
+			return authConfig, errors.Wrap(err, "could not load UserDefined template")
+		}
+		return authConfig, nil
+	}
+	err := utils.ConvertToStructuredResource(authConfigTemplateAnonymous, &authConfig)
+	if err != nil {
+		return authConfig, errors.Wrap(err, "could not load Anonymous template")
+	}
+	return authConfig, nil
+}
+
+type configMapTemplateLoader struct {
+	client   client.Client
+	fallback AuthConfigTemplateLoader
+}
+
+func NewConfigMapTempalteLoader(client client.Client, fallback AuthConfigTemplateLoader) AuthConfigTemplateLoader {
+	return &configMapTemplateLoader{
+		client:   client,
+		fallback: fallback,
+	}
+}
+
+func (c *configMapTemplateLoader) Load(authType AuthType, key types.NamespacedName) (authorinov1beta2.AuthConfig, error) {
+	// TOOD: check "authconfig-template" CM in key.Namespace to see if there is a "spec" to use, construct a AuthConfig object
```

Comment:
maybe open a issue to not lost track of it.

Reply:
https://github.com/opendatahub-io/odh-model-controller/pull/130/commits/9086d691f96b408942fc80727bc7327233bfa483